climada/hazard/landslide.py

```diff
@@ -188,304 +127,170 @@ def __init__(self):
-
-    def _get_hist_events(self, bbox, coolr_path):
-        """for LS_MODEL[0]: load gdf with landslide event POINTS from
-        global landslide catalog (COOLR of NASA) for bbox of interest"""
-        ls_gdf = geopandas.read_file(coolr_path)
-        ls_gdf_bbox = ls_gdf.cx[bbox[3]:bbox[1], bbox[2]:bbox[0]]
-        return ls_gdf_bbox
-
-    def set_ls_model_hist(self, bbox, path_sourcefile, check_plots=1):
-        """set LS from historical records documented in the NASA COOLR initiative
-        Parameters:
-            bbox (array): [N, E , S, W] for which LS hazard should be calculated.
-            path_sourcefile (str): path to shapefile with COOLR data, retrieved previously as
-                described in tutorial
-        Returns:
-            Landslide() module: LS hazard set, historic
+        self : climada.hazard.Landslide instance
+            probabilistic LS hazard
+            
+        See also
+        --------
+        sample_events_from_probs()
         """
-        if not bbox:
-            LOGGER.error('Empty bounding box, please set bounds.')
-            raise ValueError()
 
-        if not path_sourcefile:
-            LOGGER.error('No sourcefile, please specify one containing historic LS points')
-            raise ValueError()
-
-        ls_gdf_bbox = self._get_hist_events(bbox, path_sourcefile)
-
-        self.centroids.set_lat_lon(ls_gdf_bbox.latitude, ls_gdf_bbox.longitude)
-        n_cen = ls_gdf_bbox.latitude.size  # number of centroids
-        n_ev = n_cen
-        self.intensity = sparse.csr_matrix(np.ones((n_ev, n_cen)))
-        self.units = 'm/m'
-        self.event_id = np.arange(n_ev, dtype=int)
-        self.orig = np.zeros(n_ev, bool)
-        self.frequency = np.ones(n_ev) / n_ev
-        self.fraction = self.intensity.copy()
-        self.fraction.data.fill(1)
+        # raster with occurrence probs  by default stored in self.intensity:
+        self.set_raster([path_sourcefile],
+                        geometry=[shapely.geometry.box(*bbox, ccw=True)])
+
+        # sample events from probabilities
+        self.fraction = sample_events_from_probs(self.intensity/corr_fact,
+                                                  n_years)
+            
+        # set frequ. to no. of occurrences per cell / timespan:
+        self.frequency = self.fraction/n_years
+        
+        # set intensity to 1 wherever an occurrence:
+        self.intensity = self.fraction.copy()
+        self.intensity[self.intensity.nonzero()]=1       
+    
+        # meaningless, such that check() method passes:
+        self.date = np.array([])
+        self.event_name = []
+
+        self.centroids.set_raster_file(path_sourcefile,
+                                       geometry=[shapely.geometry.box(
+                                           *bbox, ccw=True)])        
+        if "unnamed" in self.centroids.crs.wkt:
+            self.centroids.meta['crs'] = {'init': 'epsg:4326', 'no_defs': True}
```

Comment:
Is this compatible with the latest version of the develop branch (aka march release?)

Reply:
i changed this syntax. self.centroids.meta['crs'] now evaluates to CRS.from_epsg(4326). is this ok? it is read into this format when using the util.coordinates function read_raster

Reply:
hmmm... in `util.coordinates` I find the line `CRS.from_user_input(DEF_CRS)`. Wouldn't this be more suitable (also thinking of future applications with flexible crs choices)?

Reply:
this just explicitly sets the crs in case it is `None` in the file that is read in. Aka it just "blindly" sets the crs to EPSG:4326

Reply:
But that is the point. The default use in CLIMADA is `DEF_CRS`. Why not us this default instead of implicitly defining a new one here?

Reply:
changed this now